pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -849,8 +853,12 @@ func (b *Build) setupSecretVolumesAndArgs(secrets []corev1.Secret, filter func(s
 		}
 
 		volumeName := fmt.Sprintf(SecretTemplateName, secret.Name)
+		msgs := validation.IsDNS1123Label(volumeName)
```

Comment:
Currently, the invalid volume name fails on pod creation leading to a reconcile error. This update will catch the error earlier but, will not lead to a successfully reconcile. kpack should successfully schedule a build with the provided secrets so, instead of returning an error we should generate a valid functioning volume name.

Reply:
should we reconcile/ convert the invalid secret volume name to valid?
1) Would there be a concern that end users supplied volume name is converted to a valid name/ would users interact with volume name directly? 
2) If we change it to a valid name, should we output the new volume name?

Reply:
The volume name here is just used an in implementation detail in the build pod. So, we can name the volume anything we like without interfering with the user's experience.

Reply:
It might even make sense to use a generated volume name for each.

Reply:
I think it would make sense to use a generated volume name to prevent any possible collision if the valid name we generate is the same as one that might be generated by a different secret name